account-kit/smart-contracts/src/ma-v2/account/modularAccountV2.ts

```diff
@@ -178,16 +231,29 @@ export async function createModularAccountV2(
     }
   })();
 
+  if (!signer) {
+    return createMAv2Base({
+      source: "ModularAccountV2", // TO DO: remove need to pass in source?
+      transport,
+      chain,
+      entryPoint,
+      signerEntity,
+      deferredAction,
+      params: params as ToWebAuthnAccountParameters, // this may fail if params is not provided
+      ...accountFunctions,
+    }) as unknown as ModularAccountV2NoSigner; // TO DO: figure out when this breaks! we shouldn't have to cast
```

Comment:
```suggestion
    return (await createMAv2Base({
      source: "ModularAccountV2", // TO DO: remove need to pass in source?
      transport,
      chain,
      entryPoint,
      signerEntity,
      deferredAction,
      params: params as ToWebAuthnAccountParameters, // this may fail if params is not provided
      ...accountFunctions,
    })) as ModularAccountV2NoSigner; // TO DO: figure out when this breaks! we shouldn't have to cast
```

Reply:
So, all that I did here was await. It is much more of an issue to cast to unknown then to a value